pkg/storage/snapshot/restore.go

```diff
@@ -143,6 +143,19 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 		return 0, nil
 	}
 
+	ready, err := target.Ready()
+	if err != nil {
+		logger.Reason(err).Error("Error checking target ready")
+		return 0, ctrl.doUpdateError(vmRestoreIn, err)
+	}
+	if !ready {
+		reason := "Waiting for target to be ready"
+		updateRestoreCondition(vmRestoreOut, newProgressingCondition(corev1.ConditionFalse, reason))
+		updateRestoreCondition(vmRestoreOut, newReadyCondition(corev1.ConditionFalse, reason))
+		// try again in 5 secs
+		return 5 * time.Second, ctrl.doUpdateStatus(vmRestoreIn, vmRestoreOut)
```

Comment:
Yeah I thought about this but I wasn't sure regarding the delete of VMI, we do have an informer watching VMI deletion, the question is if it is called when there is a delete command or when the VMI is actually deleted? since in the ready check we check for existence.. and if we return and wait for 'not exist' and the queue for the deletion already occurred then we will not queue again.. wdyt?

Reply:
Delete callback is called when the object is removed from the cache. What's the issue if we queue again? Reconciler should be idempotent

Reply:
Maybe I didnt explain myself well but you answered the question, i.e
There will be a call to the delete callback once the vmi is removed from cache so that in that add to the queue the ready function should return 'true'.
I wanted to make sure we will not miss the queue when the vmi disappears and then we will not queue again and the snapshot will fail on target not ready. So if that he case I will remove the 5sec retry.